torch/csrc/jit/tensorexpr/ir_mutator.cpp

```diff
@@ -186,7 +186,22 @@ const Expr* IRMutator::mutate(const Load* v) {
 }
 
 const Expr* IRMutator::mutate(const Buf* v) {
-  return v;
+  const Var* var = v->base_handle();
+  const Var* var_new = dynamic_cast<const Var*>(var->accept_mutator(this));
+  bool any_change = var_new != var;
```

Comment:
Theoretically, `var_new` might be `nullptr` after `dynamic_cast`. We probably should handle this case - i.e. return `nullptr` (returning `Buf(var_new, dims_new, v->dtype())` would produce a partially invalid IR, which is can lead to hard to debug issues).